src/CustomerProfile.js

```diff
@@ -0,0 +1,96 @@
+import React, { useState, useEffect } from "react";
+
+const CustomerProfile = (props) => {
+  const [customerProfileData, setCustomerProfileData] = useState(null);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    setIsLoading(true);
+    fetch(`https://cyf-react.glitch.me/customers/${props.id}`)
+      .then((res) => {
+        if (!res.ok) {
+          throw new Error(`${res.status} ${res.statusText}`);
+        }
+        return res.json();
+      })
+      .then((data) => {
+        setCustomerProfileData(data);
+        setIsLoading(false);
+      })
+      .catch((error) => {
+        console.log(error);
+        setIsLoading(false);
+      });
+  }, [props.id]);
+
+  if (isLoading) {
+    return <div>Loading customer profile...</div>;
+  }
+
+  if (!customerProfileData) {
+    return null; // or render a message indicating that the customer profile is not available
+  }
```

Comment:
[no action] It's nice you're separating the different return statements and not putting them all into one big return.